## A.R.D.E 🔥🔥🔥

Integrantes:
- Agustín Pérez
- Rayén Godoy
- Dan Salgado
- Esperanza Oliva

### Contexto y motivación

El proyecto busca acercar el conocimiento sobre los exoplanetas , planetas que orbitan estrellas fuera de nuestro sistema solar, a niños y adolescentes, mediante la creación de materiales didácticos, visuales e interactivos basados en datos reales obtenidos por la página oficial:

    * URL: https://exoplanetarchive.ipac.caltech.edu/
            https://www.jpl.nasa.gov/edu/

En los últimos años, el número de exoplanetas descubiertos ha aumentado exponencialmente, y su estudio abre nuevas preguntas sobre la habitabilidad, la composición y la diversidad de mundos en nuestra galaxia. Sin embargo, gran parte de esta información se encuentra en bases de datos científicas complejas, poco accesibles para el público general.

Este proyecto tiene como motivación transformar datos científicos en conocimiento comprensible, fomentando la curiosidad de la astronomía desde edades tempranas.

### Preguntas objetivo

* ¿Cuántos exoplanetas potencialmente habitables han sido identificados?

* ¿Cómo ha evolucionado el número de descubrimientos de exoplanetas a lo largo de los años?

* ¿Qué sistemas planetarios tienen más planetas detectados?

* ¿Cuál es la distancia promedio de los exoplanetas respecto a la Tierra según su tipo?

* ¿Existen diferencias notables en el tamaño o tipo de exoplanetas?

* ¿Que exoplanetas poseen características o condiciones similares a la tierra?


### Analisis exploratorio de datos

In [64]:
import pandas as pd
import numpy as np

path = '../data/PS_2025.10.17_07.44.21.csv'

df = pd.read_csv(path, skiprows=292, dtype={'hd_name': str, 'hip_name': str}).set_index('rowid')
df.head(3)

pl_name hostname pl_letter    hd_name   hip_name        tic_id  \
rowid                                                                    
1      11 Com b   11 Com         b  HD 107383  HIP 60202  TIC 72437047   
2      11 Com b   11 Com         b  HD 107383  HIP 60202  TIC 72437047   
3      11 Com b   11 Com         b  HD 107383  HIP 60202  TIC 72437047   

                        gaia_dr2_id                   gaia_dr3_id  \
rowid                                                               
1      Gaia DR2 3946945413106333696  Gaia DR3 3946945413106333696   
2      Gaia DR2 3946945413106333696  Gaia DR3 3946945413106333696   
3      Gaia DR2 3946945413106333696  Gaia DR3 3946945413106333696   

       default_flag  sy_snum  ...   rowupdate  pl_pubdate  releasedate  \
rowid                         ...                                        
1                 1        2  ...  2023-09-19     2023-08   2023-09-19   
2                 0        2  ...  2014-07-23     2011-08   2014-07-23   
3                 0        2  ...  2014-05-14     2008-01   2014-05-14   

      pl_nnotes  st_nphot st_nrvc st_nspec pl_nespec pl_ntranspec pl_ndispec  
rowid                                                                         
1             2         1       2        0         0            0          0  
2             2         1       2        0         0            0          0  
3             2         1       2        0         0            0          0  

[3 rows x 288 columns]

In [22]:
# Dataframe con algunas columnas que nos sirven creo, si quieren agregar mas o quitar denle nomas - agustin
# Las columnas que elegi por ahora son:

# Datos del exoplaneta
hostname = 'hostname'  # Nombre de la estrella principal
pl_name = 'pl_name'  # Nombre del exoplaneta

# Datos orbitales
pl_orbper = 'pl_orbper'  # Periodo orbital (días)
pl_orbsmax = 'pl_orbsmax'  # Semieje mayor (UA), determina distancia media a la estrella

# Datos físicos del planeta
pl_rade = 'pl_rade'  # Radio del planeta (en tierras)
pl_masse = 'pl_masse'  # Masa del planeta (en tierras)
pl_dens = 'pl_dens'  # Densidad del planeta (g/cm^3), indica composición (rocoso/gaseoso)

# Datos de temperatura
pl_orbeccen = 'pl_orbeccen'  # Excentricidad orbital, mide estabilidad de temperatura
pl_eqt = 'pl_eqt'  # Temperatura de equilibrio (K), aproxima la temperatura superficial
pl_insol = 'pl_insol'  # Flujo de insolación [Tierra=1], indica energía recibida del sol

# Datos de la estrella (afectan energía recibida)
st_teff = 'st_teff'  # Temperatura efectiva estelar (K)
st_rad = 'st_rad'  # Radio de la estrella (en radios solares)
st_lum = 'st_lum'  # Luminosidad estelar (en log(Solar))
st_mass = 'st_mass'  # Masa de la estrella (en masas solares)

# Contexto observacional
sy_dist = 'sy_dist'  # Distancia al sistema (parsecs, distancia a la Tierra creo)
sy_snum = 'sy_snum'  # Número de estrellas en el sistema (1=solitario, >1=sistema múltiple)


In [23]:
# Dataframe con las columnas seleccionadas
df_exoplanets = df[[hostname, pl_name,
                    pl_orbper, pl_orbsmax,
                    pl_rade, pl_masse, pl_dens,
                    pl_orbeccen, pl_eqt, pl_insol,
                    st_teff, st_rad, st_lum, st_mass,
                    sy_dist, sy_snum]]

df_exoplanets.head(7)

hostname   pl_name  pl_orbper  pl_orbsmax  pl_rade  pl_masse  pl_dens  \
rowid                                                                         
1       11 Com  11 Com b  323.21000       1.178      NaN       NaN      NaN   
2       11 Com  11 Com b        NaN       1.210      NaN       NaN      NaN   
3       11 Com  11 Com b  326.03000       1.290      NaN       NaN      NaN   
4       11 UMi  11 UMi b        NaN       1.510      NaN       NaN      NaN   
5       11 UMi  11 UMi b  516.22000       1.540      NaN       NaN      NaN   
6       11 UMi  11 UMi b  516.21997       1.530      NaN       NaN      NaN   
7       14 And  14 And b  185.84000       0.830      NaN       NaN      NaN   

       pl_orbeccen  pl_eqt  pl_insol  st_teff  st_rad   st_lum  st_mass  \
rowid                                                                     
1            0.238     NaN       NaN   4874.0   13.76  1.97823     2.09   
2              NaN     NaN       NaN      NaN     NaN      NaN     2.60   
3            0.231     NaN       NaN   4742.0   19.00  2.24300     2.70   
4              NaN     NaN       NaN      NaN     NaN      NaN     1.70   
5            0.080     NaN       NaN   4340.0   24.08      NaN     1.80   
6            0.080     NaN       NaN   4213.0   29.79      NaN     2.78   
7            0.000     NaN       NaN   4813.0   11.00  1.76300     2.20   

        sy_dist  sy_snum  
rowid                     
1       93.1846        2  
2       93.1846        2  
3       93.1846        2  
4      125.3210        1  
5      125.3210        1  
6      125.3210        1  
7       75.4392        1

In [26]:
# Dataframe con las columnas para el tipo de descubrimiento y asi, pense que seria util - agustin sjjsjs

# Método de descubrimiento y contexto
discoverymethod = 'discoverymethod'  # Método de descubrimiento
discovery_year = 'disc_year'  # Año de descubrimiento
disc_refname = 'disc_refname'  # Referencia del descubrimiento
disc_pubdate = 'disc_pubdate'  # Fecha de publicación del descubrimiento
disc_locale = 'disc_locale'  # Lugar del descubrimiento
disc_facility = 'disc_facility'  # Instalación del descubrimiento
disc_telescope = 'disc_telescope'  # Telescopio del descubrimiento
disc_instrument = 'disc_instrument'  # Instrumento del descubrimiento

# Métodos de detección (flags)
rv_flag = 'rv_flag'  # Detectado por variaciones de velocidad radial
pul_flag = 'pul_flag'  # Detectado por variaciones de tiempo de púlsar
ptv_flag = 'ptv_flag'  # Detectado por variaciones de tiempo de pulsación
tran_flag = 'tran_flag'  # Detectado por tránsitos
ast_flag = 'ast_flag'  # Detectado por variaciones astrométricas
obm_flag = 'obm_flag'  # Detectado por modulaciones de brillo orbital
micro_flag = 'micro_flag'  # Detectado por microlente
etv_flag = 'etv_flag'  # Detectado por variaciones de tiempo de eclipses
ima_flag = 'ima_flag'  # Detectado por imágenes
dkin_flag = 'dkin_flag'  # Detectado por cinemática de disco

soltype = 'soltype'  # Tipo de solución (confirmado, candidato, falso positivo)


In [27]:
df_discovery = df[[discoverymethod, discovery_year,
                   disc_refname, disc_pubdate, disc_locale,
                   disc_facility, disc_telescope, disc_instrument,
                   rv_flag, pul_flag, ptv_flag, tran_flag,
                   ast_flag, obm_flag, micro_flag, etv_flag,
                   ima_flag, dkin_flag, soltype]]

df_discovery.head(7)

discoverymethod  disc_year  \
rowid                               
1      Radial Velocity       2007   
2      Radial Velocity       2007   
3      Radial Velocity       2007   
4      Radial Velocity       2009   
5      Radial Velocity       2009   
6      Radial Velocity       2009   
7      Radial Velocity       2008   

                                            disc_refname disc_pubdate  \
rowid                                                                   
1      <a refstr=LIU_ET_AL__2008 href=https://ui.adsa...      2008-01   
2      <a refstr=LIU_ET_AL__2008 href=https://ui.adsa...      2008-01   
3      <a refstr=LIU_ET_AL__2008 href=https://ui.adsa...      2008-01   
4      <a refstr=DOLLINGER_ET_AL__2009 href=https://u...      2009-10   
5      <a refstr=DOLLINGER_ET_AL__2009 href=https://u...      2009-10   
6      <a refstr=DOLLINGER_ET_AL__2009 href=https://u...      2009-10   
7      <a refstr=SATO_ET_AL__2008 href=https://ui.ads...      2008-12   

      disc_locale                           disc_facility  \
rowid                                                       
1          Ground                        Xinglong Station   
2          Ground                        Xinglong Station   
3          Ground                        Xinglong Station   
4          Ground  Thueringer Landessternwarte Tautenburg   
5          Ground  Thueringer Landessternwarte Tautenburg   
6          Ground  Thueringer Landessternwarte Tautenburg   
7          Ground       Okayama Astrophysical Observatory   

                      disc_telescope             disc_instrument  rv_flag  \
rowid                                                                       
1                   2.16 m Telescope  Coude Echelle Spectrograph        1   
2                   2.16 m Telescope  Coude Echelle Spectrograph        1   
3                   2.16 m Telescope  Coude Echelle Spectrograph        1   
4      2.0 m Alfred Jensch Telescope  Coude Echelle Spectrograph        1   
5      2.0 m Alfred Jensch Telescope  Coude Echelle Spectrograph        1   
6      2.0 m Alfred Jensch Telescope  Coude Echelle Spectrograph        1   
7                   1.88 m Telescope  HIDES Echelle Spectrograph        1   

       pul_flag  ptv_flag  tran_flag  ast_flag  obm_flag  micro_flag  \
rowid                                                                  
1             0         0          0         0         0           0   
2             0         0          0         0         0           0   
3             0         0          0         0         0           0   
4             0         0          0         0         0           0   
5             0         0          0         0         0           0   
6             0         0          0         0         0           0   
7             0         0          0         0         0           0   

       etv_flag  ima_flag  dkin_flag              soltype  
rowid                                                      
1             0         0          0  Published Confirmed  
2             0         0          0  Published Confirmed  
3             0         0          0  Published Confirmed  
4             0         0          0  Published Confirmed  
5             0         0          0  Published Confirmed  
6             0         0          0  Published Confirmed  
7             0         0          0  Published Confirmed

### Datos

Los datos se obtienen de la URL: https://exoplanetarchive.ipac.caltech.edu/cgi-bin/TblView/nph-tblView?app=ExoTbls&config=PS usando TAP , descargando los datos de la tabla en formato CSV. Son datos estructurados, compuestos por:

* Tipos de datos :
    * float64(235), int64(25), object(28)
* 38973 filas, que corresponden a 38973 entradas de exoplanetas (posiblemente de un mismo planeta algunas ya que esta informacion es obtenida atravez de diferentes misiones, en las cuales se pudo haber divisado el mismo planeta)
* 288 columnas, que corresponden a los atributos encontrados de cada exoplaneta